lib/logging/lib/logging.c

```diff
@@ -140,23 +143,31 @@ log_impl(
     *timer = now + log_squelch_ns;
 
     if (err) {
+        int16_t ctx;
         char buf[256];
 
         merr_strerror(err, buf, sizeof(buf));
 
-        rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d %s: %s: %s (%d)\n",
-            file, lineno, func, fmt, buf, merr_errno(err));
+        ctx = merr_ctx(err);
+        if (ctx == 0 || !log_err_ctx_stringify) {
+            rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls), "[HSE] %s:%d: %s: %s: %s (%d)\n",
+                file, lineno, func, fmt, buf, merr_errno(err));
+        } else {
+            rc = snprintf(log_buffer_tls, sizeof(log_buffer_tls),
+                "[HSE] %s:%d: %s: %s: %s (%d): %s (%d)\n",
+                file, lineno, func, fmt, buf, merr_errno(err), log_err_ctx_stringify(ctx), ctx);
+        }
```

Comment:
Please put the format string on a line by itself in the first snprint call, much easier to read and compare these lines...